libs/utils/addCENNZnetToMetaMask.ts

```diff
@@ -1,21 +1,34 @@
-import { CENNZnetNetwork } from "@/libs/types";
 import { NETWORKS } from "@/libs/constants";
+import { MetaMaskInpageProvider } from "@metamask/providers";
+import { CENNZnetNetwork } from "@/libs/types";
 
-export default async function addCENNZnetToMetaMask(network: CENNZnetNetwork) {
-	await global.ethereum.request({
-		method: "wallet_addEthereumChain",
-		params: [
-			{
-				chainId: NETWORKS[network].chainId,
-				blockExplorerUrls: ["https://uncoverexplorer.com"],
-				chainName: NETWORKS[network].chainName,
-				nativeCurrency: {
-					name: "CPAY",
-					symbol: "CPAY",
-					decimals: 18,
-				},
-				rpcUrls: [NETWORKS[network].rpcUrl],
-			},
-		],
-	});
+export default async function addCENNZnetToMetaMask(
+	extension: MetaMaskInpageProvider,
+	network: CENNZnetNetwork
+) {
+	try {
+		await extension.request({
+			method: "wallet_switchEthereumChain",
+			params: [{ chainId: NETWORKS[network].chainId }],
+		});
+	} catch (error) {
+		if (error.code === 4902) {
```

Comment:
need to re-throw other error here, otherwise everything will be silenced.

should be `if(error?.code !== 4902) throw error;` then you can have the `4902` error handler be outside of the if block